posix.mak

```diff
@@ -2,15 +2,26 @@ INSTALL_DIR=$(PWD)/../install
 ECTAGS_LANGS = Make,C,C++,D,Sh
 ECTAGS_FILES = src/*.[chd] src/backend/*.[chd] src/root/*.[chd] src/tk/*.[chd]
 
-.PHONY: all clean test install auto-tester-build auto-tester-test
+.PHONY: all clean test install auto-tester-build auto-tester-test toolchain-info
 
 all:
 	$(QUIET)$(MAKE) -C src -f posix.mak
 
-auto-tester-build:
+auto-tester-build: toolchain-info
 	$(QUIET)$(MAKE) -C src -f posix.mak auto-tester-build ENABLE_RELEASE=1
 
-auto-tester-test: test
+auto-tester-test: toolchain-info test
+
+toolchain-info:
+	@echo '==== Toolchain Information ===='
+	${MAKE} --version
+	${SHELL} --version || true
+	uname -a
+	c++ --version
+	ld -v
```

Comment:
You're right, we're using HOST_CXX as C++ compiler. It's not currently possible to choose the linker via make's LD variable, we're simply using the C++ compiler driver to link, and wiring up `-fuse-ld=gold|bfd` requires to change dmd's annoying hardcoded link commands (a bit out of scope for _this_ PR).

Also addressed your other points.